app/facades/books/index_facade.rb

```diff
@@ -19,14 +21,43 @@ def books
       @books ||= pagination.last
     end
 
+    def categories
+      Book.categories
+    end
+
+    def ratings
+      Rating.all
+    end
+
     private
 
     def pagination
-      @pagination ||= pagy(all_books, page:)
+      @pagination ||= category.presence ? pagy_searchkick(all_books, page:) : pagy(all_books, page:)
```

Comment:
you check `category.presence` in all_books method, do you also need to check this here?